src/coreclr/jit/codegencommon.cpp

```diff
@@ -8034,6 +8117,20 @@ void CodeGen::genStructReturn(GenTree* treeNode)
         GenTreeLclVar* lclNode = actualOp1->AsLclVar();
         LclVarDsc*     varDsc  = compiler->lvaGetDesc(lclNode);
         assert(varDsc->lvIsMultiRegRet);
+#ifdef TARGET_LOONGARCH64
+        var_types type  = retTypeDesc.GetReturnRegType(0);
+        regNumber toReg = retTypeDesc.GetABIReturnReg(0);
+        GetEmitter()->emitIns_R_S(ins_Load(type), emitTypeSize(type), toReg, lclNode->GetLclNum(), 0);
+        if (regCount > 1)
+        {
+            assert(regCount == 2);
+            int offset = genTypeSize(type);
+            type       = retTypeDesc.GetReturnRegType(1);
+            offset     = (int)((unsigned int)offset < genTypeSize(type) ? genTypeSize(type) : offset);
+            toReg      = retTypeDesc.GetABIReturnReg(1);
+            GetEmitter()->emitIns_R_S(ins_Load(type), emitTypeSize(type), toReg, lclNode->GetLclNum(), offset);
+        }
+#else
```

Comment:
Here is different with the below for LA-ABI.
I think this should be kept for LA and at least different with the below.

Thanks~

Reply:
Aha, I see it now, the reuse of `type` makes it tricky to discern...

The case of interest is `genTypeSize(regOneType) < genTypeSize(regTwoType)`:
```c
// All possible combinations of sizes for the "type"s:
// struct { 4, [0 pad], 4 } - loop works
// struct { 4, [4 pad], 8 } - loop doesn't work
// struct { 8, [0 pad], 4 } - loop works
// struct { 8, [0 pad], 8 } - loop works
```
This is because in `InitializeStructReturnType` we initialize the register types for `struct { int, double }`-like constructions as `{ INT, DOUBLE }`, while for the loop to work it'd have to be `{ LONG, DOUBLE }`.

SystemV ABI has the same problem (it similarly supports passing `struct { int, double }` in the `rax/xmm0` register pair), and the way it solves it is by making `struct { int, double }` into a `{ LONG, DOUBLE }` `ReturnTypeDesc`. It seems it would be a good idea to make LA consistent with SystemV here and initialize the register types accordingly ("has 8 byte fields ?" => use `TYP_DOUBLE/getJitGCType`, else `TYP_FLOAT/TYP_INT`).